src/deadline/houdini_submitter/python/deadline_cloud_for_houdini/_assets.py

```diff
@@ -33,16 +33,27 @@ def _get_asset_references(rop_node: hou.Node) -> AssetReferences:
     return asset_references
 
 
-def _parse_files(node):
-    asset_references = _get_scene_asset_references(node)
+def _parse_files(node: hou.Node):
+    auto_detected_asset_references = _get_scene_asset_references(node)
+    assets = _get_asset_references(node)
+
     for ref in ("input_filenames", "input_directories", "output_directories"):
-        p = node.parm(ref)
-        while p.multiParmInstancesCount():
-            p.removeMultiParmInstance(0)
-        paths = sorted(list(getattr(asset_references, ref)))
-        p.set(len(paths))
-        for i, n in enumerate(p.multiParmInstances()):
-            n.set(paths[i])
+
+        auto_detected_paths = getattr(auto_detected_asset_references, ref)
+        manual_paths = getattr(assets, ref)
+        manual_paths.difference_update(auto_detected_paths)
```

Comment:
I think this logic will leave extra files in some cases. If someone has a large graph, clicks "Parse files", then removes some nodes and clicks "Parse files" again, won't the assets for the previously attached nodes remain even though they were automatically added and no longer needed? It sounded like the other submitters kept separate lists of auto-added and manually added attachments. Could we do the same?

Reply:
Right, can we not leverage these two lists as hidden parms on the node and use that to populate the UI separately?

Reply:
You're correct I believe. I'll look into adding the hidden parms to avoid that case which isn't covered in the unit tests.